From 5056133a656a4962225e4357fc2996bb404ebb65 Mon Sep 17 00:00:00 2001
From: Bang Nguyen <bangnguyen@motorola.com>
Date: Tue, 18 Aug 2015 17:04:13 -0500
Subject: [PATCH 028/959] IKKRNBSP-3334: arm/dts: 8952: TDI 1080P display
 bringup

Change-Id: I92af42d05a3005272b14d0de7f4f08b2e0707465
Signed-off-by: Bang Nguyen <bangnguyen@motorola.com>
Reviewed-on: http://gerrit.mot.com/777601
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Joseph Swantek <jswantek@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 .../dsi-panel-mot-tdi-546-1080p-cmd-common.dtsi    | 97 ++++++++++++++++++++++
 .../dts/qcom/dsi-panel-mot-tdi-546-1080p-cmd.dtsi  | 17 ++++
 arch/arm/boot/dts/qcom/msm8952-athene-p0.dts       | 12 +++
 arch/arm/boot/dts/qcom/msm8952-athene.dtsi         | 61 ++++++++++++++
 arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi    | 23 +++++
 5 files changed, 210 insertions(+)
 create mode 100644 arch/arm/boot/dts/qcom/dsi-panel-mot-tdi-546-1080p-cmd-common.dtsi
 create mode 100644 arch/arm/boot/dts/qcom/dsi-panel-mot-tdi-546-1080p-cmd.dtsi

diff --git a/arch/arm/boot/dts/qcom/dsi-panel-mot-tdi-546-1080p-cmd-common.dtsi b/arch/arm/boot/dts/qcom/dsi-panel-mot-tdi-546-1080p-cmd-common.dtsi
new file mode 100644
index 00000000000..403201f7a78
--- /dev/null
+++ b/arch/arm/boot/dts/qcom/dsi-panel-mot-tdi-546-1080p-cmd-common.dtsi
@@ -0,0 +1,97 @@
+/* Copyright (c) 2015, The Linux Foundation. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 and
+ * only version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+compatible = "qcom,mdss-dsi-panel";
+qcom,mdss-dsi-panel-name = "mipi_mot_cmd_tdi_1080p_546_v0";
+qcom,mdss-dsi-panel-supplier = "tdi";
+qcom,mdss-dsi-panel-type = "dsi_cmd_mode";
+qcom,mdss-dsi-panel-framerate = <60>;
+qcom,mdss-dsi-virtual-channel-id = <0>;
+qcom,mdss-dsi-stream = <0>;
+qcom,mdss-dsi-panel-width = <1080>;
+qcom,mdss-dsi-panel-height = <1920>;
+qcom,mdss-pan-physical-width-dimension = <68>;
+qcom,mdss-pan-physical-height-dimension = <121>;
+
+qcom,mdss-dsi-panel-clockrate = <807549120>;
+qcom,mdss-dsi-h-front-porch = <24>;
+qcom,mdss-dsi-h-back-porch = <40>;
+qcom,mdss-dsi-h-pulse-width = <4>;
+qcom,mdss-dsi-h-sync-skew = <0>;
+qcom,mdss-dsi-v-back-porch = <16>;
+qcom,mdss-dsi-v-front-porch = <16>;
+qcom,mdss-dsi-v-pulse-width = <2>;
+qcom,mdss-dsi-h-left-border = <0>;
+qcom,mdss-dsi-h-right-border = <0>;
+qcom,mdss-dsi-v-top-border = <0>;
+qcom,mdss-dsi-v-bottom-border = <0>;
+qcom,mdss-dsi-bpp = <24>;
+qcom,mdss-dsi-underflow-color = <0xff>;
+qcom,mdss-dsi-border-color = <0>;
+qcom,mdss-dsi-on-command = [
+	15 01 00 00 00 00 02 FF 26 /* Touch improvements start */
+	15 01 00 00 00 00 02 31 30
+	15 01 00 00 00 00 02 52 30
+	15 01 00 00 00 00 02 6c 30
+	15 01 00 00 00 00 02 82 30
+	15 01 00 00 00 00 02 FB 01
+	15 01 00 00 00 00 02 FF 10 /* Touch improvements end */
+	15 01 00 00 00 00 02 FF 23 /* CABC config start */
+	15 01 00 00 00 00 02 13 FF
+	15 01 00 00 00 00 02 14 F5
+	15 01 00 00 00 00 02 15 ED
+	15 01 00 00 00 00 02 16 E5
+	15 01 00 00 00 00 02 FB 01
+	15 01 00 00 00 00 02 39 95
+	15 01 00 00 00 00 02 07 00
+	15 01 00 00 00 00 02 08 02
+	15 01 00 00 00 00 02 46 B8
+	15 01 00 00 00 00 02 FF 10 /* CABC config end */
+	05 01 00 00 64 00 01 11
+	15 01 00 00 00 00 02 36 03
+	39 01 00 00 00 00 03 44 07 08
+	15 01 00 00 00 00 02 35 00
+	15 01 00 00 00 00 02 51 CC
+	15 01 00 00 00 00 02 53 2C
+	15 01 00 00 00 00 02 55 01
+	05 01 00 00 00 00 01 29];	/* display on */
+qcom,mdss-dsi-off-command = [
+	05 01 00 00 0a 00 01 28		/* display_off */
+	05 01 00 00 78 00 01 10];	/* enter sleep mode */
+
+qcom,mdss-dsi-on-command-state = "dsi_lp_mode";
+qcom,mdss-dsi-off-command-state = "dsi_lp_mode";
+qcom,mdss-dsi-h-sync-pulse = <0>;
+qcom,mdss-dsi-traffic-mode = "burst_mode";
+qcom,mdss-dsi-bllp-eof-power-mode;
+qcom,mdss-dsi-bllp-power-mode;
+qcom,mdss-dsi-lane-0-state;
+qcom,mdss-dsi-lane-1-state;
+qcom,mdss-dsi-lane-2-state;
+qcom,mdss-dsi-lane-3-state;
+qcom,mdss-dsi-panel-timings = [d6 32 22 00 60 66 26 36 28 03 04 00];
+qcom,mdss-dsi-t-clk-post = <0x1b>;
+qcom,mdss-dsi-t-clk-pre = <0x28>;
+qcom,mdss-dsi-bl-min-level = <1>;
+qcom,mdss-dsi-bl-max-level = <255>;
+qcom,mdss-dsi-bl-on-defer-delay = <40>;
+qcom,mdss-dsi-dma-trigger = "trigger_sw";
+qcom,mdss-dsi-mdp-trigger = "none";
+qcom,mdss-dsi-bl-pmic-control-type = "bl_ctrl_wled";
+qcom,mdss-dsi-reset-sequence = <0 11>, <1 11>, <0 1>, <1 11>;
+qcom,mdss-dsi-lp11-init;
+qcom,mdss-dsi-tx-eot-append;
+
+qcom,mdss-dsi-te-pin-select = <1>;
+qcom,mdss-dsi-te-dcs-command = <1>;
+qcom,mdss-dsi-te-check-enable;
+qcom,mdss-dsi-te-using-te-pin;
diff --git a/arch/arm/boot/dts/qcom/dsi-panel-mot-tdi-546-1080p-cmd.dtsi b/arch/arm/boot/dts/qcom/dsi-panel-mot-tdi-546-1080p-cmd.dtsi
new file mode 100644
index 00000000000..7d614651616
--- /dev/null
+++ b/arch/arm/boot/dts/qcom/dsi-panel-mot-tdi-546-1080p-cmd.dtsi
@@ -0,0 +1,17 @@
+/* Copyright (c) 2015, The Linux Foundation. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 and
+ * only version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+&mdss_mdp {
+	mot_tdi_546_1080p_cmd_v0: qcom,mdss_dsi_mot_tdi_546_1080p_cmd_v0 {
+		#include "dsi-panel-mot-tdi-546-1080p-cmd-common.dtsi"
+	};
+};
diff --git a/arch/arm/boot/dts/qcom/msm8952-athene-p0.dts b/arch/arm/boot/dts/qcom/msm8952-athene-p0.dts
index 74bae27bf56..46fb55fee73 100644
--- a/arch/arm/boot/dts/qcom/msm8952-athene-p0.dts
+++ b/arch/arm/boot/dts/qcom/msm8952-athene-p0.dts
@@ -18,3 +18,15 @@
 / {
 	qcom,board-id = <0x41 0x8000>;
 };
+
+&pm8950_l6 {
+	regulator-always-on;
+};
+
+&lab_regulator {
+	regulator-always-on;
+};
+
+&ibb_regulator {
+	regulator-always-on;
+};
diff --git a/arch/arm/boot/dts/qcom/msm8952-athene.dtsi b/arch/arm/boot/dts/qcom/msm8952-athene.dtsi
index 143dc142f9a..e1aca172949 100644
--- a/arch/arm/boot/dts/qcom/msm8952-athene.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8952-athene.dtsi
@@ -12,9 +12,70 @@
 
 
 #include "msm8952-moto-common.dtsi"
+#include "dsi-panel-mot-tdi-546-1080p-cmd.dtsi"
 
 / {
 	model = "Athene";
 	compatible = "qcom,msm8952-athene", "qcom,msm8952-moto",
 			"qcom,msm8952";
 };
+
+&mdss_dsi0 {
+	qcom,dsi-pref-prim-pan = <&mot_tdi_546_1080p_cmd_v0>;
+	vddio_disp-supply = <&pm8950_l6>;
+	vsp_disp-supply = <&lab_regulator>;
+	vsn_disp-supply = <&ibb_regulator>;
+};
+
+&labibb {
+	status = "ok";
+	qpnp,qpnp-labibb-mode = "lcd";
+};
+
+&ibb_regulator {
+	qcom,qpnp-ibb-discharge-resistor = <32>;
+};
+
+&dsi_panel_pwr_supply {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	qcom,panel-supply-entry@0 {
+		reg = <0>;
+		qcom,supply-name = "vddio_disp";
+		qcom,supply-min-voltage = <1800000>;
+		qcom,supply-max-voltage = <1800000>;
+		qcom,supply-enable-load = <300000>;
+		qcom,supply-disable-load = <100>;
+		qcom,supply-pre-on-sleep = <0>;
+		qcom,supply-post-on-sleep = <1>;
+		qcom,supply-pre-off-sleep = <0>;
+		qcom,supply-post-off-sleep = <1>;
+	};
+
+	qcom,panel-supply-entry@1 {
+		reg = <1>;
+		qcom,supply-name = "vsp_disp";
+		qcom,supply-min-voltage = <4600000>;
+		qcom,supply-max-voltage = <6000000>;
+		qcom,supply-enable-load = <100000>;
+		qcom,supply-disable-load = <100>;
+		qcom,supply-pre-on-sleep = <0>;
+		qcom,supply-post-on-sleep = <2>;
+		qcom,supply-pre-off-sleep = <0>;
+		qcom,supply-post-off-sleep = <5>;
+	};
+
+	qcom,panel-supply-entry@2 {
+		reg = <2>;
+		qcom,supply-name = "vsn_disp";
+		qcom,supply-min-voltage = <4600000>;
+		qcom,supply-max-voltage = <6000000>;
+		qcom,supply-enable-load = <100000>;
+		qcom,supply-disable-load = <100>;
+		qcom,supply-pre-on-sleep = <0>;
+		qcom,supply-post-on-sleep = <2>;
+		qcom,supply-pre-off-sleep = <0>;
+		qcom,supply-post-off-sleep = <1>;
+	};
+};
diff --git a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
index 3d318af3151..41b4023719e 100644
--- a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
@@ -120,3 +120,26 @@
 		};
 	};
 };
+
+&mdss_mdp {
+	qcom,mdss-pref-prim-intf = "dsi";
+};
+
+&pmx_mdss_te {
+	qcom,num-grp-pins = <1>;
+	qcom,pins = <&gp 24>;
+};
+
+&pmx_mdss {
+	qcom,num-grp-pins = <1>;
+	qcom,pins = <&gp 0>;
+};
+
+&mdss_dsi0 {
+	pinctrl-names = "mdss_default", "mdss_sleep";
+	pinctrl-0 = <&mdss_dsi_active &mdss_te_active>;
+	pinctrl-1 = <&mdss_dsi_suspend &mdss_te_suspend>;
+
+	qcom,platform-reset-gpio = <&msm_gpio 0 0>;
+	qcom,platform-te-gpio = <&msm_gpio 24 0>;
+};
-- 
2.11.0

